repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/lineage/LineagePreProcessor.java

```diff
@@ -0,0 +1,424 @@
+            Object propertyValue = vertex.getProperty(PARENT_CONNECTION_PROCESS_QUALIFIED_NAME, Object.class);
+            if (propertyValue instanceof String) {
+                return Arrays.asList((String) propertyValue);
+            } else if (propertyValue instanceof List) {
+                return (List<String>) propertyValue;
+            } else if (propertyValue != null) {
+                return Collections.singletonList(propertyValue.toString());
+            } else {
+                return Collections.emptyList();
+            }
+        } catch (Exception e) {
+            return Collections.emptyList();
+        }
+    }
+
+    private AtlasEntity createConnectionProcessEntity(Map<String, Object> connectionProcessInfo) throws AtlasBaseException {
+        AtlasEntity processEntity = new AtlasEntity(CONNECTION_PROCESS_ENTITY_TYPE);
+        processEntity.setAttribute(NAME, connectionProcessInfo.get("connectionProcessName"));
+        processEntity.setAttribute(QUALIFIED_NAME, connectionProcessInfo.get("connectionProcessQualifiedName"));
+        processEntity.setAttribute(HAS_LINEAGE, true);
+
+        AtlasObjectId inputConnection = new AtlasObjectId(CONNECTION_ENTITY_TYPE, mapOf(QUALIFIED_NAME, connectionProcessInfo.get("input")));
+        AtlasObjectId outputConnection = new AtlasObjectId(CONNECTION_ENTITY_TYPE, mapOf(QUALIFIED_NAME, connectionProcessInfo.get("output")));
+
+        Map<String, Object> relationshipAttributes = new HashMap<>();
+        relationshipAttributes.put("inputs", Collections.singletonList(inputConnection));
+        relationshipAttributes.put("outputs", Collections.singletonList(outputConnection));
+        processEntity.setRelationshipAttributes(relationshipAttributes);
+
+        try {
+            RequestContext.get().setSkipAuthorizationCheck(true);
+            AtlasEntity.AtlasEntitiesWithExtInfo processExtInfo = new AtlasEntity.AtlasEntitiesWithExtInfo();
+            processExtInfo.addEntity(processEntity);
+            EntityStream entityStream = new AtlasEntityStream(processExtInfo);
+            entityStore.createOrUpdate(entityStream, false);
+
+            checkAndUpdateConnectionLineage(inputConnection, true);
+            checkAndUpdateConnectionLineage(outputConnection, true);
+        } finally {
+            RequestContext.get().setSkipAuthorizationCheck(false);
+        }
+
+        return processEntity;
+    }
+
+    @Override
+    public void processDelete(AtlasVertex vertex) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processDeleteLineageProcess");
+
+        try {
+            Set<String> connectionProcessQNs = getConnectionProcessQNs(vertex);
+            if (connectionProcessQNs.isEmpty()) {
+                return;
+            }
+
+            for (String connectionProcessQn : connectionProcessQNs) {
+                if (!checkIfChildProcessExistForConnectionProcess(connectionProcessQn)) {
```

Comment:
If checkIfChildProcessExistForConnectionProcess returns GUIDs, you can avoid further getEntityVertex by atlasObjectId which does not use vertex cache, where GUID lookup with cache